openfe/setup/molecule.py

```diff
@@ -0,0 +1,30 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+
+from openfe.utils.molhashing import hashmol
+
+
+class Molecule:
```

Comment:
I know I asked for a Molecule class, but what if we called this `NetworkNode`, since that's likely all we use it for.  Then `node.molecule` giving the rdkit Molecule is less awful than `molecule.molecule`

Reply:
My thought was that, once we have `Molecule` class, we can use properties to "export" to different codes: `mol.rdkit`, `mol.oechem`, etc. Might also allow "import" as well.

If the user doesn't create the `Molecule` instance for `ligand`, then we don't have `(ligand in network.nodes) is (ligand in list(network.nodes))`, which is the consistency check I'm concerned about.